File: csharp/exercises/Webshop.Service/Payment/ThirdPartyPaymentIntegration.cs
using Webshop.Service.Common;

namespace Webshop.Service.Payment;

public class ThirdPartyPaymentIntegration
{
    private readonly ThirdPartyPaymentSystemClient _client;

    public ThirdPartyPaymentIntegration()
    {
        _client = new ThirdPartyPaymentSystemClient();
    }

    public bool HandlePaymentForUser(string userId, decimal amountInNok, string message)
    {
        try
        {
            _client.WaitForConnection();
            return _client.HandlePayment(userId, amountInNok, message);
        }
        catch (Exception ex)
        {
            Logger.LogError($"Payment failed: {ex.Message}");
            return false;
        }
    }
}

File: java/exercises/src/main/java/Webshop/Service/Payment/ThirdPartyPaymentIntegration.java
package Webshop.Service.Payment;

import Webshop.Service.Common.Logger;
import Webshop.Service.Payment.ThirdPartyPaymentSystemClient;

public class ThirdPartyPaymentIntegration {
    private final ThirdPartyPaymentSystemClient client;

    public ThirdPartyPaymentIntegration() {
        this.client = new ThirdPartyPaymentSystemClient();
    }

    public boolean handlePaymentForUser(String userId, double amountInNok, String message) {
        try {
            client.waitForConnection();
            return client.handlePayment(userId, amountInNok, message);
        } catch (Exception ex) {
            Logger.logError("Payment failed: " + ex.getMessage());
            return false;
        }
    }
}